### PR TITLE
Cleanup flow-parser package

### PR DESCRIPTION
**Pull request by zertosh**
tl;dr: zero deps, and only 3 files:

```
1.6K README.md
317K flow_parser.js
731B package.json
```

Compared to 2974 files before (tests, deps, etc.)

* Remove `.npmignore` in favor of `files` field in the `package.json`.
* Don't include `flowparse` and `flowvalidate` anymore.
  - Move deps used by these scripts to devDeps..
* Add `hompage`, update `repository`, and sort `package.json` (by order in https://docs.npmjs.com/files/package.json)
* Added `make npm-pack` to create a tarball of what would get published to npm (for testing purposes).

**Test plan:**

```
$ cd flow/src/parser
$ make js
$ npm pack
$ tar xf flow-parser-0.42.0.tgz
$ node -p 'require("./package").parse("1+1")'
{ type: 'Program',
  loc:
   { source: null,
     start: { line: 1, column: 0 },
     end: { line: 1, column: 3 } },
  range: [ 0, 3 ],
  body:
   [ { type: 'ExpressionStatement',
       loc: [Object],
       range: [Object],
       expression: [Object],
       directive: null } ],
  comments: [],
  errors: [] }
```

cc: @gabelevi

### CODE REVIEW

File: src/parser/README.md [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
+# The flow-parser package
 
-The Flow Parser is a JavaScript parser written in OCaml. It produces an AST that conforms to [SpiderMonkey's Parser API](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/Parser_API) and that mostly matches what [esprima](http://esprima.org/) produces. The Flow Parser can be compiled to native code or can be compiled to JavaScript using [js_of_ocaml](http://ocsigen.org/js_of_ocaml/).
+This package contains the Flow parser in its compiled-to-JavaScript form.
 
-## Building the Flow Parser
+# What is Flow
 
-Building the Flow Parser requires OCaml. Compiling to JavaScript requires js_of_ocaml >= 2.8.
+See [flowtype.org](http://flowtype.org/). The code for the Flow parser [lives on GitHub](https://github.com/facebook/flow/tree/master/src/parser).
```
Comment (mroch) [inline src/parser/README.md]: flow.org

File: src/parser/README.md [hunk 2]
```diff
@@ -1,33 +1,41 @@
...
 
-### Initial set up
+# What is the Flow Parser
```
Comment (mroch) [inline src/parser/README.md]: nit: add ? or reword as not a sentence

Comment (zertosh) [inline src/parser/README.md]: git is messing up the rename detection. this file was the old `README-npm.md` :/

**Comment by gabelevi** [comment]
Spoke to @zertosh and we're going to revert the readme rename, to keep the github functionality intact. Other than that, this looks good to me! If anyone needs the `flowparse` or `flowvalidate` scripts, I suppose they'll open an issue!

**Comment by gabelevi** [comment]
Sweet! Looks good to me!

**Pull request closed without merging**